GenerateMacroDefinitionsFile.cmake

```diff
@@ -30,6 +30,10 @@ FILE(APPEND ${GENERATED_G3_DEFINITIONS} "// CMake induced definitions below. See
 
 FOREACH(definition ${G3_DEFINITIONS} )
    FILE(APPEND ${GENERATED_G3_DEFINITIONS} "#define ${definition}\n")
+   if(definition MATCHES ".* .*")
```

Comment:
This is because we should EITHER use add_definition (rather, in fact target_compile_definitions( PUBLIC ... )) OR a generated header file.  Like this they will be defined once in the generated header and once on the command line through add_definition()
Assuming everyone uses CMake and properly uses the import script generated by this project, the target_compile_definitions approach has the advantage of not needing a generated file. (using add_definition, this is not the case, the information would be lost as soon as we are not in the context of our cmake project any more). 

However, the HUGE disadvantage would be that as soon as someone does NOT use the cmake import script, there is no way to find out how the library was compiled (except looking in the .cmake file) and even then, the definitions would have to be set by hand to properly use the library. 
The generated header on the other hand is installed alongside the library and just including it makes sure you use the same definitions. I believe it is the better solution.